### PR TITLE
Update analyzers and various fixes to be able to build with latest RC1 SDK

### PR DESCRIPTION
**Pull request by safern**
When installing latest SDK RC1 which comes with VS Internal Preview Updated  or latest P8 SDK, we're getting the net analyzers for 5.0 TFM by default, causing errors like:

```
CSC : error CS8034: Unable to load Analyzer assembly C:\Users\chucks\.nuget\packages\microsoft.codeanalysis.netanalyzers\3.3.0-beta3.20407.4\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll : Assembly with same name is already loaded [C:\dotnet\runtime\src\libraries\System.Linq.Expressions\src\System.Linq.Expressions.csproj]
CSC : error CS8034: Unable to load Analyzer assembly C:\Users\chucks\.nuget\packages\microsoft.codeanalysis.netanalyzers\3.3.0-beta3.20407.4\analyzers\dotnet\cs\Microsoft.CodeAnalysis.NetAnalyzers.dll : Assembly with same name is already loaded [C:\dotnet\runtime\src\libraries\System.Linq.Expressions\src\System.Linq.Expressions.csproj]
CSC : error CS8034: Unable to load Analyzer assembly C:\Users\chucks\.nuget\packages\microsoft.codeanalysis.netanalyzers\3.3.0-beta3.20407.4\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll : Assembly with same name is already loaded [C:\dotnet\runtime\src\libraries\System.Linq.Expressions\src\System.Linq.Expressions.csproj]
CSC : error CS8034: Unable to load Analyzer assembly C:\Users\chucks\.nuget\packages\microsoft.codeanalysis.netanalyzers\3.3.0-beta3.20407.4\analyzers\dotnet\cs\Microsoft.CodeAnalysis.NetAnalyzers.dll : Assembly with same name is already loaded [C:\dotnet\runtime\src\libraries\System.Linq.Expressions\src\System.Linq.Expressions.csproj]
```

Update to latest analyzers which with latest SDK fixes this issue.

Also, disable Producing reference assemblies as latest SDK enables them by default for net5.0 tfm.

The newest SDK no longer defines `_ShortFrameworkIdentifier` and `_ShortFrameworkVersion` which we depend for our AssemblySearchDirectories in order to support named reference in our projects. Without this we get: 

```
C:\Program Files\dotnet\sdk\5.0.100-rc.1.20414.5\Microsoft.Common.CurrentVersion.targets(2085,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "System.Collections". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [C:\repos\runtime\src\libraries\Microsoft.Win32.SystemEvents\src\Microsoft.Win32.SystemEvents.csproj]
```

cc: @stephentoub @jeffhandley @dotnet/runtime-infrastructure

### CODE REVIEW

**Comment by stephentoub** [comment]
I thought @mavasani recently made a change to do this automatically?

**Comment by mavasani** [comment]
Yes, that is corect - done recently in https://github.com/dotnet/roslyn-analyzers/pull/3974. Moving to latest `Microsoft.CodeAnalysis.NetAnalyzers` should resolve this - if not please let me know so I can fix the NuGet package.

**Comment by mavasani** [comment]
https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet5&package=Microsoft.CodeAnalysis.NetAnalyzers&protocolType=NuGet&version=5.0.0-rc1.20413.4&view=versions

**Comment by mavasani** [comment]
Also note that there was an additional follow-up SDK change that was done for this: https://github.com/dotnet/sdk/pull/12839/commits/49bf28d177a9f199913401c4939800be539864ae. So you might need both the latest SDK and NuGet package for things to fall in place.

**Comment by safern** [comment]
Thanks @mavasani -- will update the PR to move to latest analyzers and also disable them for ref assemblies since it doesn't make sense to run them there.

**Comment by stephentoub** [comment]
Thanks

**Comment by safern** [comment]
I updated the title and the descriptions as more fixes are needed to build with the latest SDK.

File: src/libraries/Directory.Build.props
Comment (jeffhandley) [inline src/libraries/Directory.Build.props:232]: Incredibly nit-picky: There are extra spaces at the end of this line.

**Review by jeffhandley — APPROVED** [review]
LGTM but please wait for another approval from someone else too. 😄

File: src/libraries/Directory.Build.props
Comment (safern) [reply src/libraries/Directory.Build.props:232]: Arghh, I thought I removed it. Thanks for catching it. 

I'm also that nit-picky with the unnecessary spaces 😄

**Review by trylek — APPROVED** [review]
LGTM

File: eng/targetingpacks.targets
Comment (ViktorHofer) [inline eng/targetingpacks.targets:12]: @safern why was this change necessary?

Comment (safern) [reply eng/targetingpacks.targets:12]: Because these properties are not longer defined on the latest SDK they were removed

Comment (ViktorHofer) [reply eng/targetingpacks.targets:12]: Thanks, I will get rid of these private properties.

**Pull request merged**
